pkg/controllerutils/miscellaneous.go

```diff
@@ -71,6 +71,7 @@ func RemoveTasks(annotations map[string]string, tasksToRemove ...string) {
 // RemoveAllTasks removes the ShootTasks annotation from the passed map.
 func RemoveAllTasks(annotations map[string]string) {
 	delete(annotations, common.ShootTasks)
+	// TODO: remove in a future release
 	delete(annotations, common.ShootTasksDeprecated)
```

Comment:
why not remove it?

Reply:
I guess it is better to have this deletion to prevent orphan annotation (that could create confusion for user or operators).